drivers/gpio/gpio_litex.c

```diff
@@ -18,6 +19,27 @@
 
 #include <soc.h>
 
+BUILD_ASSERT(CONFIG_LITEX_CSR_DATA_WIDTH == 32, "CONFIG_LITEX_CSR_DATA_WIDTH must be 32 bits");
+
+#define GPIO_LITEX_ALL_HAS_IRQ DT_ALL_INST_HAS_PROP_STATUS_OKAY(interrupts)
+#define GPIO_LITEX_ANY_HAS_IRQ DT_ANY_INST_HAS_PROP_STATUS_OKAY(interrupts)
+
+#define GPIO_LITEX_ALL_HAS_OE DT_ALL_INST_REG_HAS_NAME_STATUS_OKAY(oe)
+#define GPIO_LITEX_ANY_HAS_OE DT_ANY_INST_REG_HAS_NAME_STATUS_OKAY(oe)
+
+#define GPIO_LITEX_ALL_HAS_OUT DT_ALL_INST_REG_HAS_NAME_STATUS_OKAY(out)
+#define GPIO_LITEX_ANY_HAS_OUT DT_ANY_INST_REG_HAS_NAME_STATUS_OKAY(out)
+
+#define GPIO_LITEX_ALL_HAS_IN DT_ALL_INST_REG_HAS_NAME_STATUS_OKAY(in)
+#define GPIO_LITEX_ANY_HAS_IN DT_ANY_INST_REG_HAS_NAME_STATUS_OKAY(in)
+
+#define GPIO_LITEX_HAS_OE                                                                          \
+	(GPIO_LITEX_ANY_HAS_OE || (GPIO_LITEX_ALL_HAS_OE || gpio_config->oe_addr != 0))
+#define GPIO_LITEX_HAS_OUT                                                                         \
+	(GPIO_LITEX_ANY_HAS_OUT || (GPIO_LITEX_ALL_HAS_OUT || gpio_config->out_addr != 0))
```

Comment:
Just thinking it might be a bit better to be explicit about local variables used in macros (here and elsewhere)
```suggestion
#define GPIO_LITEX_HAS_OUT(cfg)                                                                         \
	(GPIO_LITEX_ANY_HAS_OUT && ((cfg)->out_addr != 0)))
```